learning_resources/views.py

```diff
@@ -376,6 +378,125 @@ class ResourceListItemsViewSet(NestedViewSetMixin, viewsets.ReadOnlyModelViewSet
     ordering = ["position", "-child__last_modified"]
 
 
+@extend_schema(
+    parameters=[
+        OpenApiParameter(
+            name="id",
+            type=OpenApiTypes.INT,
+            location=OpenApiParameter.PATH,
+            description="id of the learning resource",
+        ),
+        OpenApiParameter(
+            name="userlist_id",
+            type=OpenApiTypes.INT,
+            many=True,
+            location=OpenApiParameter.QUERY,
+            description="id of the parent user list",
+        ),
+        OpenApiParameter(
+            name="learning_path_id",
+            type=OpenApiTypes.INT,
+            many=True,
+            location=OpenApiParameter.QUERY,
+            description="id of the parent learning path",
+        ),
+    ],
+)
+class LearningResourceListRelationshipViewSet(viewsets.GenericViewSet):
+    """
+    Viewset for managing relationships between Learning Resources
+    and User Lists / Learning Paths
+    """
+
+    permission_classes = (AnonymousAccessReadonlyPermission,)
+    filter_backends = [MultipleOptionsFilterBackend]
+    queryset = LearningResourceRelationship.objects.select_related("parent", "child")
+    http_method_names = ["patch"]
+
+    def get_serializer_class(self):
```

Comment:
Do you know if `get_serializer_class` matters here? I would expect not since both of your methods call a specific serializer + have DRF-spectacular annotations. (More of a django question for my own education.)

Reply:
Without it you get this error during the OpenAPI spec check:

```
/home/runner/work/mit-open/mit-open/learning_resources/views.py:381: Error [LearningResourceListRelationshipViewSet]: exception raised while getting serializer. Hint: Is get_serializer_class() returning None or is get_queryset() not working without a request? Ignoring the view for now. (Exception: 'LearningResourceListRelationshipViewSet' should either include a `serializer_class` attribute, or override the `get_serializer_class()` method.)
```